### ISSUE TITLE
Shortcuts on mac os x don't work

### ISSUE THREAD

**Issue by boris1dev** [godotengine/godot#23548]
**Godot version:**
3.0.6

**OS/device including version:**
mac os x, "High Sierra", Version 10.13.6 (17G2307)

**Issue description:**
I'm facing the issue that some shortcuts on my mac don't work though they are configured in the editor settings as shortcuts.

Examples of working and not working shortcuts:: 

ctrl+cmd+D - toggle fullscreen: works

ctrl+cmd+D - switch to distraction free view: does not work

shift+cmd+O - open scene: works

cmd+y - restore after undo: works

cmd+y - custom shortcut to delete line: does not work, though the default behaviour for cmd-y (restore) has been removed

**Steps to reproduce:**
Open the script editor and press the hotkeys

**Comment by akien-mga** [comment]
Could you test 3.1 alpha 2? There were quite a few changes to macOS shortcuts in the master branch already: https://godotengine.org/article/dev-snapshot-3-1-alpha-2

**Comment by boris1dev** [comment]
Thanks for the hint.
I've tested 3.1 alpha 2
Indeed, shortcuts are working _better_: cmd+left/right also works in the editors form fields, cmd+backspace deletes at least the current line in the text editor.

cmd+ctrl+d still doesn't open the distraction free mode, cmd+backspace works in the text editor only and so on ...

**Comment by boris1dev** [comment]
**Godot version:**
3.1.alpha.official

**OS/device including version:**
mac os x, "High Sierra", Version 10.13.6 (17G2307)

**Issue description:**
I've created a probably incomplete list of all the shortcuts that I expect to work but that don't work:

1. I set as shortcut to delete a line "command+y":
![image](https://user-images.githubusercontent.com/5287874/48258194-caefeb80-e414-11e8-8224-b49c06d698ea.png)
"command+y" is only assigned to this action:
![image](https://user-images.githubusercontent.com/5287874/48258288-068ab580-e415-11e8-8ff7-5a1b6c45bc53.png)

This doesn't work. But "shift+command+k" works

2. "control-command-d" does not enable or disable the distraction free mode

3. "command+backspace" does not delete in all form fields the line. Example the search bar in the editor settings: I cannot delete "open" by placing the cursor to the right of "open" and pressing "command+backspace"
![image](https://user-images.githubusercontent.com/5287874/48259410-24a5e500-e418-11e8-85ea-0bdfe8c73820.png)

4. "alt-1" from inside the script editor doesn't switch to the scene view but inserts instead the special char "¡". 
To be honest, I'm not sure if this is a bug or feature... and it's for me not critical, as I could map it to "command+1"

**Comment by Calinou** [comment]
> "command+backspace" does not delete in all form fields the line. Example the search bar in the editor settings: I cannot delete "open" by placing the cursor to the right of "open" and pressing "command+backspace"

In Godot, <kbd>Option + Backspace</kbd> is already implemented to remove a word, but <kbd>Command + Backspace</kbd> seems to be missing (tested on commit https://github.com/godotengine/godot/commit/9c3ddf05cb9c59817d885e9daca6e8f61c89dc97).

Behavior in macOS apps like the Finder is as follows:
- <kbd>Command + Backspace</kbd> deletes everything before the cursor in the field,
- <kbd>Option + Backspace</kbd> deletes the word before the cursor in the field.

**Comment by hazarek** [comment]
command+comma doesn't work if i click script window, because there is a conflict
<img width="739" alt="cap" src="https://user-images.githubusercontent.com/36481442/89540947-64287d80-d806-11ea-99b5-9fb8afd6a414.png">

**Comment by BigZaphod** [comment]
Just conforming that In Godot 3.2.2 on macOS 10.15.6 the Alt-1,2,3 keys don't work in the script editor on macOS. It just inserts a weird character instead of switching editors. It might be better to remap those to CMD-1,2,3 for macOS default. I dunno.

**Comment by akien-mga** [comment]
This might have been fixed by #43662 or related PRs, CC @EricEzaM.

**Comment by EricEzaM** [comment]
Yeah, please test on latest master if you can build it and let us know. Some of the text editor shortcut stuff may also be fixed in #43663 (not merged yet)

**Comment by iamjoeysox** [comment]
Still having issues with this, it gets annoying when trying to quickly change between script and 2d/3d editor modes. How would I go about fixing this myself and contribute? Any ideas where I should look first? Thanks!

**Comment by iamjoeysox** [comment]
Any updates on this? Tried 3.5 and issues still persists.

**Comment by ingie** [comment]
Control-Command-D (Distraction Mode) not working on 3.5.1 on macOS 10.15.6
Changed it to Control-Command-F10 and that worked, but interestingly, editing the shortcut and trying to assign Control-Command-D says that I'm only pressing "Control-Command" even when I'm pressing D at the same time. 

This _seems_ to be a conflict with the default macOS function of looking up the selected word in the dictionary, for which Ctrl-Cmd-D is the default, however I don't seem to be able to disable that (possibly a macOS bug) as the item in Settings->Keyboard->Shortcuts->Services->Search->Look Up in Dictionary is set to "none" (despite it still working, say, in a notepad)
Even unticking the Look Up in Dictionary option, doesn't disable this feature (at the os level)

In short, I think using Control-Command-D as _the default_ on macOS seems fraught with danger, but assigning the Distraction Free mode, to say, Control-Command-F11 (to be more in line with the Windows default in Godot) works fine.

related to https://github.com/godotengine/godot/issues/22883

**Comment by passivestar** [comment]
Godot v4.1.1.stable - macOS 13.5.2

Command-backspace still doesn't work in input fields, only text editor
Command-comma still doesn't work when focus is in the text editor

**Comment by lukaskrepel** [comment]
Godot v4.1.2 - macOS v13.4.1

These shortcuts also don't work:

Distraction mode:

- Ctrl + Cmd + D

Imo, by default this should be Cmd + Shift + D (which does work), as the Ctrl/Cmd combination is not used often and is hard to do on a keyboard.

Close tab:
- Cmd + W

Next/previous tab:

- Cmd + Tab
- Cmd + Shift + Tab

By default these should really not use Cmd, because it's used for switching applications in the OS. Like Safari etc, they should default to:

- Ctrl + Tab
- Ctrl + Shift + Tab

(Which unfortunately also don't work if you set them like that in Godot)

**Comment by Calinou** [comment]
> Imo, by default this should be Cmd + Shift + D (which does work)

This is already used to duplicate lines in the script editor.

**Comment by ewersp** [comment]
`Cmd + B` is also an unfortunate conflict on macOS (Sonoma 14.2.1) with Godot (v4.2.1.stable.official).

For the Script Text Editor it is defaulted to `Go to Next Bookmark` however it is also `Run Project` from the Editor, which is a much more frequently used command in my opinion that should work while editing scripts.

**Comment by tcowley** [comment]
> `Cmd + B` is also an unfortunate conflict on macOS (Sonoma 14.2.1) with Godot (v4.2.1.stable.official).
> 
> For the Script Text Editor it is defaulted to `Go to Next Bookmark` however it is also `Run Project` from the Editor, which is a much more frequently used command in my opinion that should work while editing scripts.

I agree. I'm using MacOS Sonoma 14.2.1, and switched the bookmark prev/next shortcuts from `cmd+B` / `cmd+shift+B` to `option+B` / `option+shift+B` (those have no conflicts with other shortcuts). This frees up `cmd+B` to work as expected to run the default scene.

**Comment by mikerodd** [comment]
There is also an issue with ctrl + arrow, Godot doesn't detect it, actions in my project are not detected, at least on MacOS 10.13
Alt + Arrow and cmd + Arrow works but not Ctlr + Arrow
